tornadox/blocks/sqrtutil.py

```diff
@@ -0,0 +1,99 @@
+"""Utility functions for square-root implementations."""
+
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import jax.scipy as jsp
+
+
+@jax.jit
+def correct_noisefree(*, h, c_sqrtm):
+    return correct_noisefree_matfree(h_mul_c_sqrtm=h @ c_sqrtm, c_sqrtm=c_sqrtm)
+
+
+@jax.jit
+def correct_noisefree_matfree(*, h_mul_c_sqrtm, c_sqrtm):
```

Comment:
why the distinction with "matfree"? that's not obvious to me

Reply:
for this correction, we use "H @ c_sqrtm" to start with. So naturally, this would imply a function f(h, c_sqrtm). But sometimes, e.g. when projections are implemented via slicing, there is no matrix "h". Instead, we make a function f(g=h @ c_sqrtm) where the matmul is computed outside.

(The non-matfree version is a bit of legacy tbh. We can discuss removing it)

Reply:
Ah ok, so instead of passing H and C, we pass H*C and C, since H never appears on its own? Sounds reasonable